integration-tests/src/mock_network/setup.rs

```diff
@@ -97,6 +128,124 @@ pub fn setup_mock_network(
+    // client start height
+    let mode = if let MockNetworkMode::ProduceNewBlocks(None) = mode {
+        MockNetworkMode::ProduceNewBlocks(client_start_height.clone())
+    } else {
+        mode
+    };
+
+    // set up client dir to be ready to process blocks from client_start_height
+    if let Some(start_height) = client_start_height {
+        info!(target:"mock_network", "Preparing client data dir to be able to start at the specified start height {}", start_height);
+        let mut chain_store =
+            ChainStore::new(client_runtime.get_store(), config.genesis.config.genesis_height);
+        let mut chain_store_update = ChainStoreUpdate::new(&mut chain_store);
+        let mut network_chain_store =
+            ChainStore::new(mock_network_runtime.get_store(), config.genesis.config.genesis_height);
+
+        let network_tail_height = network_chain_store.tail().unwrap();
+        let network_head_height = network_chain_store.head().unwrap().height;
+        info!(target:"mock_network",
+              "network data chain tail height {} head height {}",
+              network_tail_height,
+              network_head_height,
+        );
+        assert!(
+            start_height <= network_head_height && start_height >= network_tail_height,
+            "client start height {} is not within the network chain range [{}, {}]",
+            start_height,
+            network_tail_height,
+            network_head_height
+        );
+        let hash = network_chain_store.get_block_hash_by_height(start_height).unwrap();
+        if !mock_network_runtime.is_next_block_epoch_start(&hash).unwrap() {
+            panic!("start height must be the last block of an epoch");
+        }
+
+        // copy chain info
+        chain_store_update
+            .copy_chain_state_as_of_block(
+                &hash,
+                mock_network_runtime.clone(),
+                &mut network_chain_store,
+            )
+            .unwrap();
+        chain_store_update.commit().unwrap();
+        info!(target:"mock_network", "Done preparing chain state");
+
+        // copy epoch info
+        let mut epoch_manager = EpochManager::new_from_genesis_config(
+            client_runtime.get_store(),
+            &config.genesis.config,
+        )
+        .unwrap();
+        let mut mock_epoch_manager = EpochManager::new_from_genesis_config(
+            mock_network_runtime.get_store(),
+            &config.genesis.config,
+        )
+        .unwrap();
+        let header = network_chain_store.get_block(&hash).unwrap().header();
+        epoch_manager
+            .copy_epoch_info_as_of_block(
+                &hash,
+                header.epoch_id(),
+                header.next_epoch_id(),
+                &mut mock_epoch_manager,
+            )
+            .unwrap();
+        info!(target:"mock_network", "Done preparing epoch info");
+
+        // copy state for all shards
```

Comment:
Wouldn't this potentially take a very long time?

Reply:
Yes, but that's unavoidable. I generally run this once to set up the directory and then just use the same directory to avoid copying state again